src/sdk/python/rtdip_sdk/pipelines/sources/spark/iso/pjm_daily_pricing_iso.py

```diff
@@ -0,0 +1,199 @@
+
+    API:              <a href="https://api.pjm.com/api/v1/">  (must be a valid apy key from PJM)
+
+    Real-Time doc:    <a href="https://dataminer2.pjm.com/feed/da_hrl_lmps/definition">
+
+    Day-Ahead doc:    <a href="https://dataminer2.pjm.com/feed/rt_hrl_lmps/definition">
+    
+    Parameters:
+        spark (SparkSession): Spark Session instance
+        options (dict): A dictionary of ISO Source specific configurations (See Attributes table below)
+
+    Attributes:
+        api_key (str): Must be a valid key from PJM, see api url
+        load_type (str): Must be one of `real_time` or `day_ahead`
+
+    Please check the BaseISOSource for available methods.
+
+    BaseISOSource:
+        ::: src.sdk.python.rtdip_sdk.pipelines.sources.spark.iso.base_iso
+    """
+
+    spark: SparkSession
+    spark_schema = PJM_PRICING_SCHEMA
+    options: dict
+    iso_url: str = "https://api.pjm.com/api/v1/"
+    query_datetime_format: str = "%Y-%m-%d %H:%M"
+    required_options = ["api_key", "load_type"]
+    default_query_timezone = "US/Eastern"
+
+    def __init__(self, spark: SparkSession, options: dict) -> None:
+        super().__init__(spark, options)
+        self.spark: SparkSession = spark
+        self.options: dict = options
+        self.load_type: str = self.options.get("load_type", "").strip()
+        self.api_key: str = self.options.get("api_key", "").strip()
+        self.days: int = self.options.get("days", 3)
+
+    def _fetch_paginated_data(self, url_suffix: str, start_date: str, end_date: str) -> bytes:
+        """
+        Fetches data from the PJM API with pagination support.
+
+        Args:
+            url_suffix: String to be used as suffix to ISO URL.
+            start_date: Start date for the data retrieval.
+            end_date: End date for the data retrieval.
+
+        Returns:
+            Raw content of the data received.
+        """
+        headers = {"Ocp-Apim-Subscription-Key": self.api_key}
+        items = []
+        query = {
+            "startRow": "1",
+            "rowCount": "50000",
+            "datetime_beginning_ept":f"{start_date}to{end_date}"
+        }
+        query_s = "&".join(["=".join([k, v]) for k, v in query.items()])
+        base_url = f"{self.iso_url}{url_suffix}?{query_s}"
+
+        next_page = base_url
+        
+        logging.info(
+            f"Requesting URL - {base_url}, start_date={start_date}, end_date={end_date}, load_type={self.load_type}"
+        )
+
+        while next_page:
+            now = datetime.now()
+            logging.info(f"Timestamp: {now}")
+            response = requests.get(next_page, headers=headers)
+            response.raise_for_status()
+            data = response.json() 
+            logging.info(f"Data for page {next_page}:")
+            items.extend(data['items'])
+            next_page = next((link['href'] for link in data['links'] if link['rel'] == 'next'), None)
```

Comment:
This may produce unexpected results as next is used for iterators and the iterator itself is being set here i.e for loop. 
If each request send a next page url , should we just not use that instead of the loop.